packages/app/src/runner/useEventManager.ts

```diff
@@ -58,6 +60,17 @@ export function useEventManager () {
     })
 
     eventManager.on('expect:origin', addCrossOriginIframe)
+
+    eventManager.on('debug:dismiss', () => {
+      const currentRoute = router.currentRoute.value
+
+      const { runId, ...query } = currentRoute.query
+
+      // Delete runId from query which will remove the test filter
+      router.replace({ ...currentRoute, query })
+
+      specStore.setActiveSpec({ ...specStore.activeSpec!, testFilter: undefined })
```

Comment:
Should we check if activeSpec is defined?

```js
if (!store.activeSpec) {
  return
}
```

We only want to call this if `activeSpec` is defined?

Also not sure `testFilter` should be grouped with `activeSpec`. It seems related, but separate. I could be convinced otherwise, what's your reasoning here?

As an aside, we've got `specFilter` and `testFilter`  which are named very similar by serve quite a different purpose. I wonder if we want a more specific name here (no suggestions right now, just a thought I had when reading this).

Is this feature only for use with the Cloud? If so, maybe `cloudDebugFilter` or `cloudFailedTests`?

Reply:
I've refactored this flow. I think `specFilter` isn't used anymore, I believe it was related to run-all-specs.